### PR TITLE
use kubeaddons-enterprise-tests repo for tests

### PR DESCRIPTION
**Pull request by zmalik**
re-enabling all tests that now reside in https://github.com/mesosphere/kubeaddons-enterprise-tests

### CODE REVIEW

**Comment by zmalik** [comment]
soft ping @mesosphere/sig-ksphere-catalog

**Comment by joejulian** [comment]
Can we be a little more generic and make that repo a single repo that can be used for kuttl tests for all kubeaddons repos? Maybe just rename it to kubeaddons-tests or something?

This is a great way to solve the problem of legacy installations.

**Comment by zmalik** [comment]
> Can we be a little more generic and make that repo a single repo that can be used for kuttl tests for all kubeaddons > repos? Maybe just rename it to kubeaddons-tests or something?

> This is a great way to solve the problem of legacy installations.

@joejulian that's a good idea but as for now that will require more changes in that repo, and its not clear who will maintain those tests. For kubeaddons-enterprise we can help but for all addons it will be hard.

File: Makefile [hunk 1]
```diff
@@ -38,8 +37,18 @@ install-bin: bin/kind bin/kubectl-kuttl
 create-kind-cluster: $(KUBECONFIG)
 
 $(KUBECONFIG): install-bin
```
Comment (faiq) [inline Makefile:39]: Does this mean that if there's no kubeconfig envvar set, it will do all of these things?

File: Makefile [hunk 2]
```diff
@@ -38,8 +37,18 @@ install-bin: bin/kind bin/kubectl-kuttl
...
-	bin/kind create cluster --wait 10s --config=test/kind/$(KUBERNETES_VERSION).yaml
+	git clone --depth 1 https://github.com/mesosphere/kubeaddons-enterprise-tests.git --branch master --single-branch
```
Comment (faiq) [inline Makefile:40]: I think this clone should be defined as a separate target

File: Makefile [hunk 3]
```diff
@@ -38,8 +37,18 @@ install-bin: bin/kind bin/kubectl-kuttl
...
+	bin/kind create cluster --wait 10s --image=kindest/node:v$(KUBERNETES_VERSION)
 
 .PHONY: kind-test
 kind-test: create-kind-cluster
```
Comment (faiq) [inline Makefile:44]: the *separate* git clone target should be a dependency here

File: Makefile [hunk 4]
```diff
@@ -38,8 +37,18 @@ install-bin: bin/kind bin/kubectl-kuttl
...
-# TODO restore once catalog is tested
+	kubeaddons-enterprise-tests/run-tests.sh
+	bin/kind delete cluster
```
Comment (faiq) [inline Makefile:46]: This seems kind of redundant if we're doing all of this in clean anyways. my suggestion is to take this out

**Comment by zmalik** [comment]
I will re-work the PR to include @joejulian suggestion to rename `kubeaddons-enterprise-tests` to `kubeaddons-tests`

**Comment by zmalik** [comment]
@joejulian this is ready for review, the test repo has been renamed to `kubeaddons-tests` and references updated

**Pull request merged**